dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/UseCallingThreadExecutor.java

```diff
@@ -49,10 +79,43 @@ static boolean shouldUseCallingThreadExecutor() {
      *
      * @param newProbability number between {@code 0.0} and {@code 1.0}.
      */
-    static void setCallingThreadExecutorProbability(float newProbability) {
+    void setCallingThreadExecutorProbability(float newProbability) {
         Preconditions.checkArgument(
                 newProbability >= DISABLED && newProbability <= ENABLED,
                 "Probability should be between 0.0 and " + "1.0");
-        UseCallingThreadExecutor.probability = newProbability;
+        probability = newProbability;
+    }
+
+    @VisibleForTesting
+    float getCurrentProbability() {
+        return probability;
+    }
+
+    static UseCallingThreadExecutor get() {
+        return INSTANCE;
+    }
```

Comment:
This `get` method looks a little odd, I'd probably try to avoid changing the code in `callBlocking`. Perhaps we split the implementation into a package-private-for-testing class which we can instantiate with custom values, and leave this class in place as a static adapter to a singleton instance of that class?

Reply:
Done